src/sparkle/selector/selector.py

```
@@ -73,11 +90,13 @@ def construct(
             The construction Run
         """
         selection_scenario.create_scenario()
-        selector = self.selector_class(
-            model_class=self.model_class,
-            budget=selection_scenario.solver_cutoff,
-            maximize=not selection_scenario.objective.minimise,
-        )
+        metadata = self.build_metadata_from_scenario(selection_scenario)
+        selector = self.instantiate_selector(metadata)
+        # Ensure selector carries a usable model_class for ASF CLI build
+        if not hasattr(selector, "model_class") or isinstance(
```

Why is this not a given by ASF? Should this not always be here after instantiating a Selector? If not, Sparkle should raise an error that the selector could not be instantiated based on the settings